src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -4456,6 +4456,166 @@ paths:
   ## SNAPSHOT BUILDER API ENDPOINTS
   ## ⚠️ NOTE: ALL SNAPSHOT APIS ARE SUBJECT TO CHANGE
   ##############################################################################
+  /api/repository/v1/snapshots/{id}/snapshotBuilder/settings:
+    get:
+      tags:
+        - snapshots
+        - repository
+      description: >
+        Get the snapshot builder settings for a snapshot.
+      operationId: getSnapshotSnapshotBuilderSettings
+      parameters:
+        - in: path
+          description: The UUID of the dataset.
+          name: id
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        200:
+          description: The snapshot builder settings
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SnapshotBuilderSettings'
+        400:
```

Comment:
> There is a [default response](https://swagger.io/docs/specification/describing-responses/) that might be useful for the variety of error conditions in this situation

Interesting, I wasn't aware of this. It does bug me that every API has to have its own copy of the usual set of error responses. Since these are all APIs still in progress anyway I can try adding them for these and see how we like it.

> Another approach I like that makes this file more manageable is to break up the [path and schema into separate files using $refs](https://swagger.io/docs/specification/using-ref/)

This is an interesting idea. The last time I looked at using `$ref` in this way, it was part of the openapi spec but the swagger-ui we were using didn't support it, leading to gradle build steps like this: https://github.com/DataBiosphere/terra-workspace-manager/blob/main/openapi/build.gradle

If this build step is no longer required and `$ref` has good IDE support then I think it would make sense to use. I can create a tech debt ticket to look into this, which project should I make the ticket in?

Reply:
@pshapiro4broad - I would imagine these kinds of things would go to our board: https://broadworkbench.atlassian.net/jira/software/c/projects/DCJ/boards/204/backlog

Reply:
OK I created https://broadworkbench.atlassian.net/browse/DCJ-258 for this